src/emulator.py

```diff
@@ -428,10 +428,27 @@ def apply_settings(
 
 def allow_unsafe() -> None:
     client = TrezorClientDebugLink(get_device())
-    # ignore for Legacy firmware, there is no such setting
-    if client.features.major_version == 1:
-        return
     client.open()
     time.sleep(SLEEP)
-    device.apply_settings(client, safety_checks=1)  # TODO
+
+    # Each model has its own supported safety checks level
+    if client.features.major_version == 1:
+        safety_checks = messages.SafetyCheckLevel.PromptTemporarily
+    else:
+        safety_checks = messages.SafetyCheckLevel.PromptAlways
+
+    # Some older devices do not support safety checks, so we know
+    # the command will fail with a specific error message
+    # T1 supports safety checks from 1.10.1 and T2 from 2.3.2
+    try:
+        device.apply_settings(client, safety_checks=safety_checks)
+    except TrezorFailure as err:
+        # Catching only specific error message, otherwise reraising the exception
+        if "No setting provided" in str(err):
+            log(
+                f"Could not allow unsafe. Device does not support safety checks. Err: {err}",
+                "red",
+            )
+        else:
+            raise
```

Comment:
This doesn't have to be `raise err`? I honestly do not know 🤔

Reply:
no :)